pkg/costmodel/costmodel.go

```diff
@@ -928,6 +940,15 @@ func (cm *CostModel) GetNodeCost(cp costAnalyzerCloud.Provider) (map[string]*cos
 	nodeList := cm.Cache.GetAllNodes()
 	nodes := make(map[string]*costAnalyzerCloud.Node)
 
+	vgpuCount, err := getAllocatableVGPUs(cm.Cache)
```

Comment:
This can be an expensive operation, I suspect you only want to compute this for nodes claiming to have a vGPU.

Reply:
AFAIK `GetNodeCost` doesn't run on every node, so this would run once and merely apply it to every node requesting a vGPU. Is that not the case? I was under the impression that this function is called only two single times (once in `GetNodeCost` and once in `ComputeCostData`) in `CostModelMetricsEmitter.Start()`.

Reply:
ComputeCostData is itself called in a loop for metrics emission.

Reply:
after sync-ing offline, this operation is probably fine so let's not prematurely optimize.